packages/grid/_modules_/grid/hooks/features/export/seralizers/csvSeraliser.ts

```diff
@@ -1,8 +1,10 @@
 import { GridColumns, GridRowId, GridRowModel } from '../../../../models';
 
-export function buildRow(row: GridRowModel, columns: GridColumns): Array<GridRowModel> {
-  const mappedRow: GridRowModel[] = [];
-  columns.forEach((column) => column.field !== '__check__' && mappedRow.push(row[column.field]));
+export function buildRow(row: GridRowModel, columns: GridColumns): Array<string> {
+  const mappedRow: string[] = [];
+  columns.forEach(
+    (column) => column.field !== '__check__' && mappedRow.push(`"${row[column.field]}"`),
```

Comment:
this doesn't care about valueGetter in colDef.
I'm adding a `getCellValue` in my editRows PR. But not exposed so maybe we should expose it.

Reply:
BTW you only need to wrap it with "" when there are special chars

Reply:
Ok, I think we need to extend the functionality a bit. I spent some time reading the specks yesterday and check how others solve that issue and here is the outcome. The problem is more complex than it seems.

Most competitors have a default functionality that works OTB with the simplest case (in our example that would be to either keep it as it is or wrap the cell values in `"`). I think that wrapping it with `"` will work for most cases. In addition they have a way for a developer to pass in their delimiter/separator to be used for formatting, meaning that if the default is `"` then if you have values with `"` in it then you can choose to use `,`. That way if the OTB functionality doesn't work for a developer's use case they can implement them themselves by calling the provided API and pass in their delimiter/separator.

On the topic of handling that case, ourselves -> I played a bit with a potential algorithm but the issue is that you need to traverse the entire data at least 2 times to make it work, which is a case of very big data will be time-consuming. In the basic case, the complexity is O(2n) but the problem is that if you have `"` and `,` and `|` then it wouldn't be possible to format it, basically, I don't think we can handle all the cases ourselves.

Reply:
Could you share the links of what you have explored so far? 

This solution seems to add double quotes for comma and escape double quotes. Would it be enough? If not, why? https://stackoverflow.com/a/24922761 not enough?

Reply:
Maybe we could salvage logic from https://github.com/jmaister/excellentexport/blob/master/src/excellentexport.ts

Reply:
I think it won't cover all the cases. The issue is that you have 3 ways to format a csv `"` or `,` or `|`, and if the data you want to format has all 3 then how would you decide which to use?

In terms of links -> https://www.ag-grid.com/documentation/vue/csv-export/ for example.

Reply:
What do you mean by "format"?

Reply:
Material Table is using https://github.com/mbrn/filefy/blob/master/src/csv-builder/CsvBuilder.ts

Reply:
The current way the CSV works is that it only supports `,` as a delimiter. I think having the option for developers to set their own delimiters through the CSV export API will work because I don't think we can handle all the cases where the data contains different potential delimiters?

Reply:
I agree we should only start by supporting ',' as it's the most common use case. Then wrap with "" and escape the value as in `escapeCell ` or `fixCSVField ` of the link shared above.

Using the proper value is more important.

Reply:
Ok, I've updated the formatter. It will now handle the case where the delimiter is `,` and if there is a `,` in a cell value that cell value will be wrapped in a `"` resulting in a correctly formatted CSV. I also added an addition to the existing test case.

Reply:
It works with the latest changes